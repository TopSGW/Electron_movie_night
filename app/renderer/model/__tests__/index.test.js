'use strict'
/* globals jest, describe, test, expect */

import {
  createReduxStore,
  getAllMovies, // public selectors
  getCrawlActive,
  getFeaturedMovie,
  getSearchCategory,
  getSearchQuery,
  getVisibleMovies
} from '../index'

jest.mock('../../../shared/utils', () => {
  return {
    isDevEnv: () => true
  }
})

// Set up some test data
const movies = [
  { title: 'Old School', director: 'Todd Phillips', actors: ['Will Ferrell', 'Luke Wilson'] },
  { title: 'Super Troopers', director: 'Jay Chand', actors: ['Kevin Hefferman', 'Steve Lemme'] },
  { title: 'Wedding Crashers', director: 'David Dobkin', actors: ['Vince Vaugh', 'Owen Wilson'] },
  { title: 'Meet The Parents', director: 'Jay Roach', actors: ['Ben Stiller', 'Robert De Niro'] }
]

const state = {
  crawl: { active: true, directory: '/path/to/some/dir' },
  featuredMovie: { movie: 'someMovie', action: 'click', panelID: 2 },
  movies: {
    displayOrder: ['thriller', 'action', 'drama'],
    movieDB: [
      { genre: 'thriller', movies: [movies[0], movies[1]] },
      { genre: 'action', movies: [movies[0], movies[1], movies[2], movies[3]] },
      { genre: 'drama', movies: [movies[0], movies[1], movies[2]] }
    ]
  },
  search: {
    category: 'action',
    query: 'Old School'
  }
}

// For the public selectors, all of the logic is really delegated to other
// selectors.  Thus, we will just take snapshots here so we can be alerted of
// any changes and make sure that is what we expected.
describe('public selectors', () => {
  test('getAllMovies', () => {
    expect(getAllMovies(state)).toMatchSnapshot()
  })
  test('getCrawlActive', () => {
    expect(getCrawlActive(state)).toMatchSnapshot()
  })
  test('getFeaturedMovie', () => {
    expect(getFeaturedMovie(state)).toMatchSnapshot()
  })
  test('getSearchCategory', () => {
    expect(getSearchCategory(state)).toMatchSnapshot()
  })
  test('getSearchQuery', () => {
    expect(getSearchQuery(state)).toMatchSnapshot()
  })
  test('getVisibleMovies', () => {
    expect(getVisibleMovies(state)).toMatchSnapshot()
  })
})

//
describe('store', () => {
  test('getState', () => {
    const store = createReduxStore()
    expect(store.getState()).toMatchSnapshot()
  })
})
