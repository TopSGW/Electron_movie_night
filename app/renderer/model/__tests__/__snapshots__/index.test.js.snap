// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`public selectors getAllMovies 1`] = `
Array [
  Object {
    "genre": "thriller",
    "movies": Array [
      Object {
        "actors": Array [
          "Will Ferrell",
          "Luke Wilson",
        ],
        "director": "Todd Phillips",
        "title": "Old School",
      },
      Object {
        "actors": Array [
          "Kevin Hefferman",
          "Steve Lemme",
        ],
        "director": "Jay Chand",
        "title": "Super Troopers",
      },
    ],
  },
  Object {
    "genre": "action",
    "movies": Array [
      Object {
        "actors": Array [
          "Will Ferrell",
          "Luke Wilson",
        ],
        "director": "Todd Phillips",
        "title": "Old School",
      },
      Object {
        "actors": Array [
          "Kevin Hefferman",
          "Steve Lemme",
        ],
        "director": "Jay Chand",
        "title": "Super Troopers",
      },
      Object {
        "actors": Array [
          "Vince Vaugh",
          "Owen Wilson",
        ],
        "director": "David Dobkin",
        "title": "Wedding Crashers",
      },
      Object {
        "actors": Array [
          "Ben Stiller",
          "Robert De Niro",
        ],
        "director": "Jay Roach",
        "title": "Meet The Parents",
      },
    ],
  },
  Object {
    "genre": "drama",
    "movies": Array [
      Object {
        "actors": Array [
          "Will Ferrell",
          "Luke Wilson",
        ],
        "director": "Todd Phillips",
        "title": "Old School",
      },
      Object {
        "actors": Array [
          "Kevin Hefferman",
          "Steve Lemme",
        ],
        "director": "Jay Chand",
        "title": "Super Troopers",
      },
      Object {
        "actors": Array [
          "Vince Vaugh",
          "Owen Wilson",
        ],
        "director": "David Dobkin",
        "title": "Wedding Crashers",
      },
    ],
  },
]
`;

exports[`public selectors getCrawlActive 1`] = `true`;

exports[`public selectors getFeaturedMovie 1`] = `
Object {
  "action": "click",
  "movie": "someMovie",
  "panelID": 2,
}
`;

exports[`public selectors getSearchCategory 1`] = `"action"`;

exports[`public selectors getSearchQuery 1`] = `"Old School"`;

exports[`public selectors getVisibleMovies 1`] = `
Array [
  Object {
    "genre": "action",
    "movies": Array [
      Object {
        "actors": Array [
          "Will Ferrell",
          "Luke Wilson",
        ],
        "director": "Todd Phillips",
        "title": "Old School",
      },
    ],
  },
]
`;

exports[`store getState 1`] = `
Object {
  "crawl": Object {
    "active": false,
    "directory": "",
    "inProgress": Array [],
    "moviesFound": 0,
    "showCrawlStatsOverlay": false,
  },
  "dbLoaded": false,
  "featuredMovie": Object {
    "action": "",
    "movie": null,
    "panelID": -1,
  },
  "movies": Object {
    "displayOrder": Array [],
    "movieDB": Array [],
  },
  "search": Object {
    "category": "",
    "query": "",
  },
}
`;
